dotnet/src/Connectors/Connectors.OpenAI/TextToImage/AzureOpenAITextToImageService.cs

```diff
@@ -35,6 +36,49 @@ public sealed class AzureOpenAITextToImageService : ITextToImageService
     /// </summary>
     public static string DeploymentNameKey => "DeploymentName";
 
+    /// <summary>
+    /// Create a new instance of Azure OpenAI image generation service
+    /// </summary>
+    /// <param name="deploymentName">Deployment name identifier</param>
+    /// <param name="endpoint">Azure OpenAI deployment URL</param>
+    /// <param name="credential">Token credentials, e.g. DefaultAzureCredential, ManagedIdentityCredential, EnvironmentCredential, etc.</param>
+    /// <param name="modelId">Model identifier</param>
+    /// <param name="httpClient">Custom <see cref="HttpClient"/> for HTTP requests.</param>
+    /// <param name="loggerFactory">The ILoggerFactory used to create a logger for logging. If null, no logging will be performed.</param>
+    /// <param name="apiVersion">Azure OpenAI Endpoint ApiVersion</param>
+    public AzureOpenAITextToImageService(
+        string deploymentName,
+        string endpoint,
+        TokenCredential credential,
+        string? modelId,
+        HttpClient? httpClient = null,
+        ILoggerFactory? loggerFactory = null,
+        string? apiVersion = null)
+    {
+        Verify.NotNull(credential);
+        Verify.NotNullOrWhiteSpace(deploymentName);
+
+        this._deploymentName = deploymentName;
+
+        if (modelId is not null)
+        {
+            this.AddAttribute(AIServiceExtensions.ModelIdKey, modelId);
+        }
+        this.AddAttribute(DeploymentNameKey, deploymentName);
+
+        this._logger = loggerFactory?.CreateLogger(typeof(AzureOpenAITextToImageService)) ?? NullLogger.Instance;
+
+        var connectorEndpoint = !string.IsNullOrWhiteSpace(endpoint) ? endpoint! : httpClient?.BaseAddress?.AbsoluteUri;
+        if (connectorEndpoint is null)
+        {
+            throw new ArgumentException($"The {nameof(httpClient)}.{nameof(HttpClient.BaseAddress)} and {nameof(endpoint)} are both null or empty. Please ensure at least one is provided.");
+        }
+
+        this._client = new(new Uri(connectorEndpoint),
+            credential,
+            GetClientOptions(httpClient, apiVersion));
+    }
```

Comment:
Can you add the following new constructor instead.

By proving a custom `OpenAIClient` this will allow the client options to also be set.

```
    /// <summary>
    /// Create a new instance of Azure OpenAI image generation service
    /// </summary>
    /// <param name="deploymentName">Deployment name identifier</param>
    /// <param name="modelId">Model identifier</param>
    /// <param name="openAIClient"><see cref="OpenAIClient"/> to use for the service.</param>
    /// <param name="loggerFactory">The ILoggerFactory used to create a logger for logging. If null, no logging will be performed.</param>
    public AzureOpenAITextToImageService(
        string deploymentName,
        string? modelId,
        OpenAIClient openAIClient,
        ILoggerFactory? loggerFactory = null)
    {
        Verify.NotNullOrWhiteSpace(deploymentName);

        this._deploymentName = deploymentName;

        if (modelId is not null)
        {
            this.AddAttribute(AIServiceExtensions.ModelIdKey, modelId);
        }
        this.AddAttribute(DeploymentNameKey, deploymentName);

        this._logger = loggerFactory?.CreateLogger(typeof(AzureOpenAITextToImageService)) ?? NullLogger.Instance;

        this._client = openAIClient;
    }
```

Reply:
Hi,
Ok, I can add it.
During my code review of other clients, I seen that ChatCompletion or embeddings used the AzureOpenAIClientCore core as private OpenAICore client :
`

    /// <summary>Core implementation shared by Azure OpenAI clients.</summary>
    private readonly AzureOpenAIClientCore _core;

    /// <summary>
    /// Create an instance of the <see cref="AzureOpenAIChatCompletionService"/> connector with API key auth.
    /// </summary>
    /// <param name="deploymentName">Azure OpenAI deployment name, see https://learn.microsoft.com/azure/cognitive-services/openai/how-to/create-resource</param>
    /// <param name="endpoint">Azure OpenAI deployment URL, see https://learn.microsoft.com/azure/cognitive-services/openai/quickstart</param>
    /// <param name="apiKey">Azure OpenAI API key, see https://learn.microsoft.com/azure/cognitive-services/openai/quickstart</param>
    /// <param name="modelId">Azure OpenAI model id, see https://learn.microsoft.com/azure/cognitive-services/openai/how-to/create-resource</param>
    /// <param name="httpClient">Custom <see cref="HttpClient"/> for HTTP requests.</param>
    /// <param name="loggerFactory">The <see cref="ILoggerFactory"/> to use for logging. If null, no logging will be performed.</param>
    public AzureOpenAIChatCompletionService(
        string deploymentName,
        string endpoint,
        string apiKey,
        string? modelId = null,
        HttpClient? httpClient = null,
        ILoggerFactory? loggerFactory = null)
    {
        this._core = new(deploymentName, endpoint, apiKey, httpClient, loggerFactory?.CreateLogger(typeof(AzureOpenAIChatCompletionService)));
        this._core.AddAttribute(AIServiceExtensions.ModelIdKey, modelId);
    }
`

So maybe it will be great if also the image generation service used this same object and same contructors ?
But there is a GetClientOptions method that override the API version :

`
    private static OpenAIClientOptions GetClientOptions(HttpClient? httpClient, string? apiVersion) =>
        ClientCore.GetOpenAIClientOptions(httpClient, apiVersion switch
        {
            // DALL-E 3 is supported in the latest API releases
            _ => OpenAIClientOptions.ServiceVersion.V2024_02_15_Preview
        });
`

So maybe that's why in the initial version the OpenAiClient was not exposed in constructor.
Maybe I can recreate a AzureOpenAIClient and override the ClientOptions ?

Reply:
I have updated my PR to use a shared constructor to reduce code duplication.
Open pPoint : Should be expose the OpenAIClient and let consumer to have a bad API Version in configuration ?